### PR TITLE
firewalld: bring back zone target set

### LINKED ISSUE #112: firewalld module missing the target feature due to the migration from community.general
##### SUMMARY

The firewalld module was moved from `community.general` to `ansible.posix` but it seems the `target` feature that was implemented in `community.general` was not moved in the process.

The PR that added the target feature in `community.general`: https://github.com/ansible-collections/community.general/pull/526

The PR that deleted firewalld from `community.general`: https://github.com/ansible-collections/community.general/pull/623
Note that this PR contains the target feature.

The PR that added firewalld to `ansible.posix`: #68
Note that this one does not include the target feature.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
firewalld

##### ANSIBLE VERSION

```paste below
ansible 2.10.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/st0rmingbr4in/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.6 (default, Sep 30 2020, 04:00:38) [GCC 10.2.0]
```

##### CONFIGURATION

```
$ ansible-config dump --only-changed | cat -e
$
```

##### OS / ENVIRONMENT

archlinux

### PR DESCRIPTION
**Pull request by St0rmingBr4in**
#### SUMMARY
Bring back the `target` feature in the firewalld module.
The feature was left out when the firewalld module was moved from `community.general` to `ansible.posix`

Fixes #112

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
firewalld

### CODE REVIEW

File: changelogs/fragments/firewalld_zone_target.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - firewalld - Bring the ``target`` feature back (Was deleted during the migration from community.general).
```
Comment (aminvakil) [inline changelogs/fragments/firewalld_zone_target.yml:2]: https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment
> Each changelog entry must contain a link to its issue between parentheses at the end. If there is no corresponding issue, the entry must contain a link to the PR itself.
```suggestion
  - firewalld - Bring the ``target`` feature back (https://github.com/ansible-collections/ansible.posix/issues/112).
```

File: plugins/modules/firewalld.py [hunk 2]
```diff
@@ -84,6 +84,13 @@
     description:
       - Whether to run this module even when firewalld is offline.
     type: bool
+  target:
+    description:
+      - firewalld Zone target
+      - If state is set to C(absent), this will reset the target to default
+    choices: [ default, ACCEPT, DROP, REJECT ]
+    type: str
+    version_added: 0.2.0
```
Comment (aminvakil) [inline plugins/modules/firewalld.py:93]: The next version would be 1.1.2 according to https://github.com/ansible-collections/ansible.posix/releases.
```suggestion
    version_added: 1.1.2
```

File: plugins/modules/firewalld.py [hunk 3]
```diff
@@ -636,10 +696,12 @@ def main():
             interface=dict(type='str'),
             masquerade=dict(type='str'),
             offline=dict(type='bool'),
+            target=dict(type='str', required=False, choices=['default', 'ACCEPT', 'DROP', 'REJECT']),
```
Comment (aminvakil) [inline plugins/modules/firewalld.py:699]: I think by default `required` is `False` and there is no need to state that.
```suggestion
            target=dict(type='str', choices=['default', 'ACCEPT', 'DROP', 'REJECT']),
```

File: plugins/modules/firewalld.py (hunk 2 above)
Comment (aminvakil) [inline plugins/modules/firewalld.py:93]: Latest released version is 1.1.1 according to https://github.com/ansible-collections/ansible.posix/tags, but I'm not sure what would be next version (1.2.0 ? 2.0.0 ? something else ?)

Maybe 1.2.0, but I'm not sure.

Comment (St0rmingBr4in) [reply plugins/modules/firewalld.py:93]: Since we are adding some functionality I would say 1.2.0

File: plugins/modules/firewalld.py (hunk 3 above)
Comment (St0rmingBr4in) [reply plugins/modules/firewalld.py]: Done

File: changelogs/fragments/firewalld_zone_target.yml (hunk 1 above)
Comment (St0rmingBr4in) [reply changelogs/fragments/firewalld_zone_target.yml]: Done

**Comment by St0rmingBr4in** [comment]
Quick question, what is ansible/gate? Why is it pending?

**Comment by St0rmingBr4in** [comment]
Oh right it's the condition on the label to have the PR automatically merged. Sorry for the question. Is there anything else preventing this to be merged?

**Comment by St0rmingBr4in** [comment]
@aminvakil can this be merged? I can make modifications if necessary. :slightly_smiling_face:

**Comment by aminvakil** [comment]
@St0rmingBr4in I'm not maintainer of this branch, I don't know if this is going to be merged or is it something that's preventing this to get merged or ...

**Comment by St0rmingBr4in** [comment]
@maxamillion could you take a look ?

**Review by maxamillion — APPROVED** [review]

**Comment by maxamillion** [comment]
this is great, thanks!

**Pull request merged**